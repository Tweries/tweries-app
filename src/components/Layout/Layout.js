import React from 'react';
import { useAuth0 } from '../../react-auth0-wrapper';
import NavBar from '../NavBar/NavBar';

function Layout() {
  const {
    isAuthenticated,
    loading,
    loginWithRedirect,
    logout,
    user
  } = useAuth0();

  return (
    <div className="container mx-auto px-4">
      <div className="flex mb-4">
        <div className="w-full">
          {loading ? (
            <h2 className="text-center">...</h2>
          ) : (
            <NavBar
              dispatch={console.log}
              isAuthenticated={isAuthenticated}
              loginWithRedirect={loginWithRedirect}
              logout={logout}
              user={user}
            />
          )}
        </div>
      </div>
      <div className="flex mb-4">
        <div className="w-full">
          <h1 className="font-bold logo my-4 text-5xl text-center">Tweries</h1>
        </div>
      </div>
    </div>
  );
}

export default Layout;
