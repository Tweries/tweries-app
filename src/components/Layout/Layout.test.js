import { render } from '@testing-library/react';
import React from 'react';
import { useAuth0 } from '../../react-auth0-wrapper';
import Layout from './Layout';

jest.mock('../../react-auth0-wrapper');

describe('Layout', () => {
  it('to match snapshot', () => {
    useAuth0.mockImplementation(() => ({}));

    const { container } = render(<Layout />);

    expect(container).toMatchSnapshot();
  });

  it('loading', () => {
    useAuth0.mockImplementation(() => ({ loading: true }));

    const { container } = render(<Layout />);

    expect(container).toMatchSnapshot();
  });
});
