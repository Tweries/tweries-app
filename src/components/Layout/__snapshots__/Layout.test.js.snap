// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinefeedPicker loading 1`] = `
<div>
  <div
    class="container mx-auto px-4"
  >
    <div
      class="flex mb-4"
    >
      <div
        class="w-full"
      >
        <h2
          class="text-center"
        >
          ...
        </h2>
      </div>
    </div>
    <div
      class="flex mb-4"
    >
      <div
        class="w-full"
      >
        <h1
          class="font-bold logo my-4 text-5xl text-center"
        >
          Tweries
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`LinefeedPicker to match snapshot 1`] = `
<div>
  <div
    class="container mx-auto px-4"
  >
    <div
      class="flex mb-4"
    >
      <div
        class="w-full"
      >
        <header
          class="flex flex-col items-center"
        >
          <button
            class="bg-gray-300 border border-gray-500 font-bold px-4 rounded"
            data-testid="login"
          >
            Log in
          </button>
        </header>
      </div>
    </div>
    <div
      class="flex mb-4"
    >
      <div
        class="w-full"
      >
        <h1
          class="font-bold logo my-4 text-5xl text-center"
        >
          Tweries
        </h1>
      </div>
    </div>
  </div>
</div>
`;
